include/buffer_tree.h

```diff
@@ -122,16 +122,18 @@ class BufferTree {
    * they should check their wait condition again
    * Useful when switching from blocking to non-blocking calls
    * to the circular queue
-   * For example: we use this when shutting down the graph_workers\
+   * For example: we set this to true when shutting down the graph_workers
+   * @param    bypass is true if we should turn on non-blocking operations
+   *           and false if we should turn them off
    * @return   nothing
    */
-  void bypass_wait();
+  void bypass_wait(bool bypass);
```

Comment:
since this function now toggles instead of sets, we should rename it to avoid confusion

Reply:
Any ideas?

Reply:
`toggle_wait`?

Reply:
I think I've going to go with set_non_block(true/false)